### PR TITLE
container: Rework API and model around "base commit" and "merge commit"

### PR DESCRIPTION
**Pull request by cgwalters**
containers: Expose a state struct with base commit and layering state

This is actually quite analogous to what rpm-ostree does internally;
a lot of the internals there distinguish "base" versus "layered"
commits, and this is very similar.

Here, when we import a non-layered image, we can mostly
ignore the merge commit since all it has is the image manifest.
The base image layer ref *is* the encapsulated ostree commit with
all the metadata injected by (rpm-)ostree in the non-layered case.

Add an API which exposes this as a struct, and also return it
from the importer's `AlreadyPresent` case.

---

container: Change import result case to contain state struct

Notably, this also stops exposing the ostree ref for the merge commit,
which I think is a good idea in general since it should be thought
of more as an implementation detail.

In other words, this module speaks container image references
and ostree commits.

---

container/deploy: Use base commit if we're not layered

If we're not doing a layered image, then use the base commit
for the deployment.

Closes: https://github.com/ostreedev/ostree-rs-ext/issues/143

---

container/deploy: Also write ref with target if provided

With this new emphasis on "dual commit objects" for the container
deployment, the higher level code queries via container image references
and we don't expose the ostree ref (since there is no longer a single
one).

This makes it critical to write the internal ref matching the
target container image, because it now needs to match the origin.

---

### CODE REVIEW

File: lib/src/container/store.rs [hunk 1]
```diff
@@ -355,6 +369,46 @@ pub fn list_images(repo: &ostree::Repo) -> Result<Vec<String>> {
         .collect()
 }
 
+fn query_image_impl(
+    repo: &ostree::Repo,
+    imgref: &OstreeImageReference,
+) -> Result<Option<(ImageManifest, LayeredImageState)>> {
```
Comment (cgwalters) [inline lib/src/container/store.rs:397]: I thought about exposing the parsed manifest in `LayeredImageState` too, but it can come later.

**Comment by cgwalters** [comment]
I've updated https://github.com/coreos/rpm-ostree/pull/3139 to use this, and https://github.com/coreos/coreos-assembler/pull/2523 is also updated and I've tested them all together.

**Review by lucab — APPROVED** [review]
